lib/frosting/repository.rb

```diff
@@ -1,18 +1,45 @@
 require "active_support/core_ext/string/inflections"
+require "active_support/core_ext/module/delegation"
 
 module Frosting
   class Repository
+    def self.infer_presenter(resource)
+      "Presenters::#{resource.class.name}".constantize
+    end
+
+    def self.procify(arg)
+      arg.respond_to?(:call) ? arg : Proc.new { arg }
```

Comment:
:water_buffalo: `Proc.new` :arrow_right: [`proc`](http://batsov.com/articles/2014/02/04/the-elements-of-style-in-ruby-number-12-proc-vs-proc-dot-new/)